dds/DCPS/unique_ptr.h

```diff
@@ -130,6 +130,11 @@ class unique_ptr {
     std::swap(ptr_, b.ptr_);
   }
 
+  bool operator<(const unique_ptr& other) const
+  {
+    return *ptr_ < *other.ptr_;
```

Comment:
I think comparing pointer values is not what we want. So `operator<` in `TypeIdentifier` relies on a number of `operator<` for member types, for example `PlainSequenceSElemDefn::operator<`, which calls `External<TypeIdentifier>::operator<` which finally calls `unique_ptr<TypeIdentifier>::operator<`. Since we're using `TypeIdentifier` as keys in some maps, if `unique_ptr<TypeIdentifier>::operator<` returns the comparison of pointer values, we may not be able to find an entry with a particular `TypeIdentifier` object value even though it's there in the map. So I think comparing using the actual contents of the objects is what we want in the case of `TypeIdentifier`.

The weakness is that all types that use `unique_ptr` now must provide `operator<`. That was a reason I thought about modifying the `operator<` operators of member types of `TypeIdentifier` directly so that they don't rely on `unique_ptr::operator<`.

Reply:
It needs to match the standard library definition. If we want the caller to compare the pointed-to objects, then that code should change.

Reply:
So should I change the operators `operator<` in `TypeIdentifier`?